lib/app.rb

```diff
@@ -5,50 +5,73 @@
-puts person.correct_name
-capitalized_person = CapitalizeDecorator.new(person)
-puts capitalized_person.correct_name
-capitalized_trimmed_person = TrimmerDecorator.new(capitalized_person)
-puts capitalized_trimmed_person.correct_name
+  # retrieve the data from the user
+  def get_chomp(arr)
+    received = {}
+    arr.each do |elt|
+      print "#{elt} :"
+      value = gets.chomp
+      received[elt] = value
+    end
+    received # return the hash
+  end
 
-class1 = Classroom.new('A1')
-class2 = Classroom.new('B1')
+  # display the main menu commande
+  def list_commands(arr)
+    arr.map.with_index { |command, index| puts "#{index + 1} - #{command}" }
+  end
 
-malick = Student.new(19, class1, 'malick')
-jacques = Student.new(17, class1, 'jacques', parent_permission: false)
-amy = Student.new(17, class1, 'amy', parent_permission: true)
-james = Teacher.new(34, 'Theories', 'james')
+  def main_header
+    main_command = [
+      'List all books', 'List all people',
+      'Create a person', 'Create a book', 'Create a rental',
+      'List all rentals for a given person id', 'Exit'
+    ]
+    puts "\n Welcomme to school library app"
+    puts " Please choose an option by enrerin a number \n"
+    list_commands(main_command)
+  end
 
-puts(class1.students.map(&:name))
-puts(class2.students.map(&:name))
-class2.add_student(amy)
-puts(class1.students.map(&:name))
-puts(class2.students.map(&:name))
-jacques.classroom = class2
-puts jacques.classroom.label
-puts(class1.students.map(&:name))
-puts(class2.students.map(&:name))
+  def execute_option(choice)
+    menu_options = {
+      '1' => :list_books,
+      '2' => :list_people,
+      '3' => :people_menu,
+      '4' => :create_book,
+      '5' => :create_rendal,
+      '6' => :list_my_rentals,
+      '7' => :exit_app
+    }
+    if menu_options[choice]
+      send(menu_options[choice]) # send invokes another method by its name
+    else
+      puts 'WRONG KEY'
+      main_menu
+    end
+  end
 
-book1 = Book.new('The great congo', 'Malumbu')
-book2 = Book.new('A short talk', 'Moses')
-book3 = Book.new('Java', 'Island')
+  def exit_app
+    abort('Thank you for using this app.')
+  end
 
-Rental.new('2017-11-11', amy, book1)
-Rental.new('2017-10-10', amy, book2)
-Rental.new('2017-10-10', jacques, book1)
-Rental.new('2017-10-10', malick, book1)
-Rental.new('2017-11-9', malick, book2)
-puts(amy.rentals.map { |rental| rental.book.title })
-puts(book2.rentals.map { |rental| rental.person.name })
-malick.add_rental(book3, '2017-11-9')
-puts(book3.rentals.map { |rental| rental.person.name })
-puts(malick.rentals.map { |rental| rental.book.title })
-puts(book1.rentals.map { |rental| rental.person.name })
-rent = Rental.new('2017-11-9', james, book2)
-book1.add_rental(rent)
-puts(book1.rentals.map { |rental| rental.person.name })
-puts(james.rentals.map { |rental| rental.book.title })
+  def main_menu
+    main_header
+    choice = gets.chomp
+    execute_option(choice)
+  end
+end
```

Comment:
- [ ] As per requirements the `app.rb` file should serve as your console app entry point. It should have methods that do the following:
- List all books.
- List all people.
- Create a person (teacher or student, not a plain Person).
- Create a book.
- Create a rental.
- List all rentals for a given person id.

Currently, this logic is being implemented in the files in the `/modules` folder. Kindly implement it in the `app.rb`